apps/vaporgui/PDoubleInput.cpp

```diff
@@ -1,21 +1,23 @@
 #include "PDoubleInput.h"
 #include "VLineItem.h"
 #include <vapor/ParamsBase.h>
-#include <QLineEdit>
+#include "VLineEdit.h"
 
 PDoubleInput::PDoubleInput(const std::string &tag, const std::string &label)
-: PLineItem(tag, _doubleInput = new VDoubleInput, label)
+: PLineItem(tag, label, _doubleInput = new VLineEdit)
 {
-//    connect(_doubleInput, SIGNAL(ValueChanged(double)), this, SLOT(doubleInputValueChanged(double)));
+    _doubleInput->SetIsDouble(true);
+    connect(_doubleInput, &VLineEdit::ValueChanged, this, &PDoubleInput::doubleInputValueChanged);
 }
 
 void PDoubleInput::updateGUI() const
 {
-    // double value = getParams()->GetValueDouble(GetTag(), 0);
-    // _doubleInput->SetValue(value);
+    double value = getParamsDouble();
+    _doubleInput->SetValue(to_string(value));
 }
 
-void PDoubleInput::doubleInputValueChanged(double v)
+void PDoubleInput::doubleInputValueChanged(const std::string &v)
 {
-    getParams()->SetValueDouble(GetTag(), GetTag(), v);
+    double d = stod(v);
```

Comment:
Since VLineEdit is the vapor widget for entering doubles I was waiting for it to have a signal that gives a double. @sgpearse do you plan on adding this, does VLineEdit always guarantee the string is a double if SetIsDouble is true, or should I add safety checks here?